### PR TITLE
feat(github): migrate the gitproviderreceiver githubscraper to githubreceiver

### PR DESCRIPTION
**Pull request by gnmeyer**
This PR separates the GitHub scraper from GitProviderReceiver into its own separate GitHubReceiver component.

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -38,6 +38,44 @@ receivers:
           static_configs:
             - targets: [0.0.0.0:8888]
 
+  github:
+    initial_delay: 10s
+    collection_interval: 60s
+    scrapers:
+      github:
+        metrics:
+          #Default Metrics (enabled by default)
```
Comment (adrielp) [inline config/config.yaml:47]: Why was this change made?

File: config/config.yaml [hunk 2]
```diff
@@ -78,6 +116,6 @@ service:
   extensions: [health_check, pprof, zpages, bearertokenauth/github]
   pipelines:
     metrics:
-      receivers: [gitprovider, prometheus]
+      receivers: [github, gitprovider, prometheus]
```
Comment (adrielp) [inline config/config.yaml:119]: This in general is just an example config, not meant to necessarily be used. We can remove what we feel is necessary, like prometheus, and gitprovider.

File: config/manifest.yaml [hunk 3]
```diff
@@ -66,5 +67,6 @@ providers:
   - gomod: go.opentelemetry.io/collector/confmap/provider/yamlprovider v0.109.0
 
 replaces:
+  - github.com/liatrio/liatrio-otel-collector/receiver/githubreceiver => ../receiver/githubreceiver/
```
Comment (adrielp) [inline config/manifest.yaml:70]: We can remove the git provider.

File: receiver/githubreceiver/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+## Scraping
+
+> Important:
+> * The GitHub scraper does not emit metrics for branches that have not had
+>   changes since creation from the default branch (trunk).
+> * Due to GitHub API limitations, it is possible for the branch time metric to
+>   change when rebases occur, recreating the commits with new timestamps.
+
+
+<!-- TODO: Combine this documentation once the scraper code is restructured due scope change -->
```
Comment (adrielp) [inline receiver/githubreceiver/README.md:88]:
```suggestion
```

File: receiver/githubreceiver/config.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	metadata.MetricsBuilderConfig  `mapstructure:",squash"`
+}
+
+var _ component.Config = (*Config)(nil)
+var _ confmap.Unmarshaler = (*Config)(nil)
+
+// Validate the configuration passed through the OTEL config.yaml
+func (cfg *Config) Validate() error {
+	if len(cfg.Scrapers) == 0 {
+		return errors.New("must specify at least one scraper")
```
Comment (adrielp) [inline receiver/githubreceiver/config.go:35]: I'm not sure if we adjusted this message, but probably should. Could be done on the next iteration.

**Review by adrielp — APPROVED** [review]

File: config/config.yaml (hunk 2 above)
Comment (gnmeyer) [reply config/config.yaml:119]: Roger, I wanted to clear up most of that

**Pull request merged**